pkg/controller/jobframework/reconciler.go

```diff
@@ -373,6 +374,16 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, err
 	}
 
+	// Update workload conditions if implemented JobWithCustomWorkloadConditions interface.
+	if jobCond, ok := job.(JobWithCustomWorkloadConditions); ok {
+		if conditions, updated := jobCond.CustomWorkloadConditions(wl); updated {
+			wlPatch := workload.BaseSSAWorkload(wl)
+			wlPatch.Status.Conditions = conditions
+			return reconcile.Result{}, r.client.Status().Patch(ctx, wlPatch, client.Apply,
+				client.FieldOwner(fmt.Sprintf("%s-custom", constants.JobControllerName)))
```

Comment:
Let's make `fmt.Sprintf("%s-custom", constants.JobControllerName))` a constant, there is no point in formatting it at runtime if nothing is changing.

Reply:
Good point, thank you.